source/jobs/JobsFeature.cpp

```diff
@@ -689,3 +703,8 @@ int JobsFeature::stop()
 
     return 0;
 }
+
+std::shared_ptr<AbstractIotJobsClient> JobsFeature::getJobsClient()
+{
+    return std::shared_ptr<AbstractIotJobsClient>(new IotJobsClientWrapper(thingName, mqttConnection));
+}
```

Comment:
One more thing.. if you don't intend to use call jobsClient from multiple classes or pass that as a parameter, wouldn't it make more sense to use std::unique_ptr?

which then makes this function become 
```suggestion
void JobsFeature::getJobsClient()
{
    jobsClient = std::unique_ptr<AbstractIotJobsClient>(new IotJobsClientWrapper(thingName, mqttConnection));
}
```

and also might want to rename getJobsClient to something like createJobsClient regardless of using unique_ptr or not since it might get confused as a simple getter method while in reality a new object is being created. Feel free to disagree though.

Reply:
The shared_ptr is only because of testing. Since you can't have two unique_ptrs pointing to the same object you can't create a Mock unique_ptr<MockIotJobsClient> and inject it into JobsFeature and then use the mock in the tests. Since unique_ptr is move only you can't keep the reference to it in the tests after injecting it... If that makes sense.